CppImport/src/macro/ExpansionManager.cpp

```diff
@@ -0,0 +1,213 @@
+		 * we check whether we are not already in a xMacro body because we want to remember the .h part (the first one) of
+		 * the xMacro begins (the potential .cpp part could overwrite the stored information otherwise).
+		 */
+		currentXMacroParent = entry;
+	else if (currentXMacroParent && !entry->parent)
+	{
+		/*
+		 * if we are in a xMacro body and the current expansion is not a top level expansion then this expansion is a
+		 * xMacro child of the stored xMacro expansion.
+		 */
+		entry->xMacroParent = currentXMacroParent;
+		currentXMacroParent->xMacroChildren.append(entry);
+	}
+
+	entry->metaCall =
+			new OOModel::MetaCallExpression(definitionManager_->definitionName(entry->definition));
+
+	if (!md->getMacroInfo()->isObjectLike()) // only function like macros have braces in their signature to parse
+	{
+		// extract everything in parentheses of the expansion signature using a regular expression
+		QRegularExpression regex ("\\((.*)\\)", QRegularExpression::DotMatchesEverythingOption);
+		auto argumentsString = lexicalHelper_->unexpandedSpelling(sr);
+		auto match = regex.match(argumentsString);
+		auto arguments = match.captured(1).split(",");
+
+		// by default initialize meta call arguments to be reference expressions with the raw spelling at this expansion
+		for (auto i = 0; i < clang_->argumentNames(entry->definition).size(); i++)
+		{
+			auto actualArg = args->getUnexpArgument((unsigned int)i);
+			entry->metaCall->arguments()->append(new OOModel::ReferenceExpression(arguments[i]));
+			entry->argumentLocs.append(actualArg->getLocation());
+		}
+	}
+
+	expansions_.append(entry);
+}
+
+QVector<MacroExpansion*> ExpansionManager::expansions() { return expansions_; }
+
+
+void ExpansionManager::clear()
+{
+	expansionCache_.clear();
+	expansions_.clear();
+}
+
+QVector<MacroExpansion*> ExpansionManager::topLevelExpansions()
+{
+	QVector<MacroExpansion*> result;
+	for (auto expansion : expansions_)
+		if (!expansion->parent)
+			result.append(expansion);
+
+	return result;
+}
+
+
+MacroExpansion*ExpansionManager::expansion(clang::SourceLocation loc)
+{
+	MacroExpansion* expansion = immediateExpansion(loc);
+	MacroExpansion* last = expansion;
+
+	if (expansion)
+	{
+		do
+		{
+			last = expansion;
+			loc = clang_->sourceManager()->getImmediateExpansionRange(loc).first;
+			expansion = immediateExpansion(loc);
+		} while (expansion && expansion->isChildOf(last));
+	}
+
+	return last;
+}
+
+MacroExpansion*ExpansionManager::immediateExpansion(clang::SourceLocation loc)
+{
+	auto expansion = clang_->immediateMacroLoc(loc);
+	for (auto i = 0; i < expansions_.size(); i++)
+		if (expansions_[i]->range.getBegin() == expansion) return expansions_[i];
+
+	/*
+	 * if we have not found an immediate expansion for loc then we try again using the found immediateExpansionLoc.
+	 * this can happen in case of token concatenation or stringifycation where the first expansion location would point
```

Comment:
it's spelled "stringification" with an i and not y. Do a quick text search to fix this everywhere, I've seen at least one other place with this mistake.